src/web/common.rkt

```diff
@@ -87,47 +87,63 @@
 (define (preprocess->c preprocess)
   (match preprocess
     [(list 'sort variables ...)
-     (format "assert(~a);" (format-less-than-condition variables))]))
+     (format "assert(~a);" (format-less-than-condition variables))]
+    [(list 'abs x)
+     (format "~a = abs(~a);" x x)]))
 
 (define (preprocess->java preprocess)
   (match preprocess
     [(list 'sort variables ...)
-     (format "assert ~a;" (format-less-than-condition variables))]))
+     (format "assert ~a;" (format-less-than-condition variables))]
+    [(list 'abs x)
+     (format "~a = Math.abs(~a);" x x)]))
 
 (define (preprocess->python preprocess)
   (match preprocess
     [(list 'sort variables ...)
      (define comma-joined (string-join (map ~a variables) ", "))
-     (format "[~a] = sort([~a])" comma-joined comma-joined)]))
+     (format "[~a] = sort([~a])" comma-joined comma-joined)]
+    [(list 'abs x)
+     (format "~a = abs(~a)" x x)]))
 
 (define (preprocess->julia preprocess)
   (match preprocess
     [(list 'sort variables ...)
       (define comma-joined (string-join (map ~a variables) ", "))
-      (format "~a = sort([~a])" comma-joined comma-joined)]))
+      (format "~a = sort([~a])" comma-joined comma-joined)]
+    [(list 'abs x)
+     (format "~a = abs(~a)" x x)]))
 
 (define (preprocess->matlab preprocess)
   (match preprocess
     [(list 'sort variables ...)
      (define comma-joined (string-join (map ~a variables) ", "))
-     (format "~a = num2cell(sort([~a])){:}" comma-joined comma-joined)]))
+     (format "~a = num2cell(sort([~a])){:}" comma-joined comma-joined)]
+    [(list 'abs x)
+     (format "~a = abs(~a)" x x)]))
 
 (define (preprocess->tex preprocess)
   (match preprocess
     [(list 'sort variables ...)
      (define comma-joined (string-join (map ~a variables) ", "))
-     (format "[~a] = \\mathsf{sort}([~a])\\\\" comma-joined comma-joined)]))
+     (format "[~a] = \\mathsf{sort}([~a])\\\\" comma-joined comma-joined)]
+    [(list 'abs x)
+     (format "~a = |~a|\\\\" x x)]))
 
 (define (preprocess->default preprocess)
-  (define note
+  (define sort-note
     "NOTE: ~a should be sorted in increasing order before calling this function.")
```

Comment:
I usually define things like this in global scope, outside the function, because then it's more obvious that it's just a constant.